# Sprint 3 - Manipulación de Datos I (Sesiones)

En el presente caso de estudio se propone una introducción práctica a los fundamentos de manipulación y análisis de datos utilizando la librería **Tidyverse**, una de las herramientas más importantes y ampliamente utilizadas dentro del ecosistema de **R** para el procesamiento de información estructurada. Ante esto, el objetivo principal de estas actividades es desarrollar habilidades iniciales para trabajar con conjuntos de datos reales, comprendiendo no solo las técnicas básicas de transformación y exploración de información, sino también la lógica metodológica que guía un proyecto de análisis de datos.

En el ámbito profesional, el análisis de datos no se limita únicamente a ejecutar código o generar gráficos; requiere comprender el problema que se desea resolver, interpretar correctamente la información disponible y estructurar adecuadamente cada etapa del proceso analítico. Por esta razón, además de introducir herramientas técnicas de manipulación de datos, este caso de estudio busca familiarizarte con algunos de los principios fundamentales que permiten organizar y desarrollar proyectos analíticos de manera ordenada, reproducible y orientada a la toma de decisiones.

Por un lado, a través de **Tidyverse** trabajarás con estructuras tabulares de información, aprendiendo a cargar, inspeccionar, limpiar, transformar y analizar datos provenientes de archivos externos. Estas capacidades técnicas constituyen la base de disciplinas como la analítica de negocios, la ciencia de datos, la inteligencia artificial y la investigación aplicada, donde la calidad del análisis depende directamente de la correcta preparación y comprensión de los datos disponibles.

De manera complementaria, durante el desarrollo de este caso se introducirán los primeros criterios metodológicos asociados a un flujo de trabajo analítico básico, abordando por ahora las siguientes etapas:

* Entendimiento del contexto.
* Entendimiento de los datos.
* Preparación de los datos.
* Análisis de datos.

Cada una de estas etapas cumple un rol fundamental dentro de un proyecto de análisis de datos. Comprender el contexto permite identificar el problema de negocio que se desea resolver; entender los datos facilita reconocer la utilidad, limitaciones, estructura y significado de los mismos; preparar los datos asegura la consistencia y calidad de la información a usarse; y finalmente, el análisis de datos permite extraer hallazgos, patrones y conclusiones útiles para apoyar procesos de decisión en el ámbito del problema definido.

Importante mencionar que el desarrollo de estas habilidades representa un primer acercamiento a la construcción de proyectos analíticos reales, promoviendo no solo competencias técnicas relacionadas con programación y manipulación de datos, sino también capacidades analíticas y metodológicas esenciales para enfrentar problemas complejos basados en información.

## Entendimiento del contexto

El ranking de selecciones masculinas de la FIFA constituye uno de los sistemas de clasificación deportiva profesional más reconocidos a nivel mundial. Este mecanismo fue implementado oficialmente en el año 1993, poco antes de la realización de la Copa Mundial de 1994, con el objetivo de establecer una metodología estandarizada que permitiera comparar el desempeño relativo de las distintas selecciones nacionales afiliadas. Desde entonces, el ranking se ha convertido en una referencia central para evaluar el rendimiento competitivo de los equipos nacionales en el contexto internacional.

Esta clasificación se actualiza de manera periódica —habitualmente con frecuencia mensual— tomando en consideración los resultados obtenidos por cada selección en partidos oficiales y amistosos reconocidos por la FIFA. Entre los encuentros considerados se incluyen competiciones continentales, eliminatorias mundialistas, torneos internacionales y partidos amistosos disputados durante las fechas oficiales establecidas en el calendario oficial. Dependiendo de la relevancia del torneo, la fortaleza del rival y el resultado obtenido, cada encuentro aporta una determinada cantidad de puntos al cálculo global de cada selección.

Más allá de su carácter informativo para aficionados, periodistas y analistas deportivos, el ranking FIFA posee una importancia estratégica significativa dentro de la organización del fútbol internacional. Actualmente, este sistema constituye uno de los principales criterios utilizados para definir los bombos y grupos en competiciones oficiales de gran relevancia, especialmente en Copa Mundial. Esto implica que una mejor posición en el ranking puede influir directamente en el nivel de dificultad que enfrentará una selección durante la fase inicial de un torneo mundialista, afectando potencialmente sus probabilidades de clasificación a rondas posteriores.

A lo largo de su historia, el ranking ha experimentado múltiples ajustes metodológicos orientados a mejorar la precisión y representatividad del sistema de puntuación. Entre las modificaciones más relevantes se encuentran las implementadas en enero de 1999, agosto de 2006, agosto de 2011 y agosto de 2018. Cada uno de estos cambios respondió a críticas y limitaciones detectadas en las versiones anteriores del modelo, particularmente relacionadas con la sobrevaloración de ciertos partidos amistosos, diferencias regionales en la competitividad y fluctuaciones excesivas en las posiciones del ranking.

La modificación más reciente, implementada en 2018, introdujo un sistema basado en modelos tipo Elo, que se utilizan en otros deportes y disciplinas competitivas. Este enfoque busca reflejar de manera más realista el desempeño de las selecciones mediante ajustes dinámicos en función de los resultados esperados y obtenidos en cada partido. Como consecuencia, el ranking actual presenta una mayor estabilidad y sensibilidad frente a la calidad de los oponentes y la importancia competitiva de cada encuentro.

Desde una perspectiva analítica, el ranking FIFA representa un caso de estudio particularmente interesante para el análisis de datos y la modelación estadística. Su evolución histórica permite explorar patrones de desempeño deportivo, cambios en la competitividad entre regiones, impacto de eventos internacionales y comportamiento temporal de las selecciones nacionales. Además, el conjunto de datos asociado al ranking ofrece oportunidades para aplicar técnicas de limpieza, transformación, análisis exploratorio y visualización de datos utilizando herramientas como **Tidyverse**.

## Entendimiento de los datos

Lo primero que te sugiero hacer al iniciar cualquier proyecto de análisis de datos es importar las librerías y módulos con los que vas a trabajar. En este caso en concreto vamos cargar **Tidyverse** cuyo objetivo es proporcionar funciones y comandos para un eficiente manejo de estructuras tabulares (también conocidos como *dataframes* o *datasets*).

In [1]:
library("tidyverse")

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors


A continuación, conviene cargar los datos en la variable `df_fifa` mediante la función `read_csv` de **Tidyverse**. En este caso usaremos la información proveniente del archivo [**fifa_rank.csv**](https://raw.githubusercontent.com/jsaraujo5081/material_curso_data_analytics/refs/heads/main/datasets/fifa_rank.csv). 

In [2]:
df_fifa <- read_csv(file = "https://raw.githubusercontent.com/jsaraujo5081/material_curso_data_analytics/refs/heads/main/datasets/fifa_rank.csv")

Rows: 57797 Columns: 6
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr  (3): Country, Confederation, Points_New_Version
dbl  (2): Points_Old_Version, rank
date (1): Rank_Date

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


Una buena práctica al momento de cargar los datos es observar sus principales características a fin de establecer un plan de tratamiento acorde al dataset y su contexto. Esto resulta de vital importancia puesto que cada conjunto de datos es único y tiene sus propias particularidades. En concreto, interesa ver al menos lo siguiente:

* Estructura de los datos (número de filas y columnas)
* Nombres de las columnas
* Tipos de variables en cada columna
* Aspectos particulares de cada columna

Con este propósito, utiliza la función `glimpse` para un primer vistazo al dataset.

In [3]:
glimpse(df_fifa)

Rows: 57,797
Columns: 6
$ Country            <chr> "Germany", "Italy", "Switzerland", "Sweden", "Argen…
$ Confederation      <chr> "UEFA", "UEFA", "UEFA", "UEFA", "CONMEBOL", "UEFA",…
$ Rank_Date          <date> 1993-08-08, 1993-08-08, 1993-08-08, 1993-08-08, 19…
$ Points_Old_Version <dbl> 57, 57, 50, 55, 51, 54, 52, 55, 49, 51, 55, 45, 49,…
$ Points_New_Version <chr> "ND", "ND", "ND", "ND", "ND", "ND", "ND", "ND", "ND…
$ rank               <dbl> 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, …


Una forma más expresa de conocer las dimensiones del dataset (filas y columnas) es mediante la función `dim`. Aplicalo en nuestros datos.

In [4]:
dim(df_fifa)

[1] 57797     6

El dataset **fifa_rank** contiene cerca de 58 mil filas y 6 columnas detalladas a continuación:

* Country: Nombre del pais afiliado a la FIFA cuya selección de fútbol es puntuada.
* Confederation: Confederación a la que pertenece de acuerdo a la división de la FIFA.
* Rank_Date: Fecha de publicación del ranking FIFA entre marzo de 1993 y junio de 2018.
* Points_Old_Version: Puntos obtenidos conforme metodologías de cálculo anteriores al 2011 
* Ponts_New_Version: Puntos obtenidos conforme la metodología posterior a 2011. 
* rank: Posición en el ranking oficial de la FIFA dados los puntos calculados.

A continuación algunos aspectos detectados:

* Los nombres de las columnas tienen mayúsculas y esto puede ser problemático en las siguientes etapas del proyecto puesto que complicarían de forma innecesaria la consulta de la información.
* El tipo de variable de Points_New_Version es *string* lo cual no hace sentido considerando que deberías ver números.

Profundiza en tu conocimiento del dataset utilizando ahora la función `sample_n` con el argumento `size = 15`, a fin de extraer una muestra aleatoria de los datos de tamaño 15.

In [5]:
sample_n(df_fifa, size = 15)

Country,Confederation,Rank_Date,Points_Old_Version,Points_New_Version,rank
<chr>,<chr>,<date>,<dbl>,<chr>,<dbl>
Laos,AFC,1998-07-15,13,ND,137
Suriname,CONCACAF,2018-01-18,NA,120.82,160
Belarus,UEFA,2012-10-03,NA,409.63,87
United Arab Emirates,AFC,2013-01-17,NA,346.36,103
Cuba,CONCACAF,2013-11-28,NA,641.13,47
Nigeria,CAF,1999-06-16,489,ND,68
Albania,UEFA,2012-03-07,NA,432.16,80
Iceland,UEFA,1996-11-20,35,ND,58
Romania,UEFA,2006-03-15,685,ND,26


Identifica en más detalle los valores perdidos en las distintas columnas. Utiliza para esto las funciones `colSums` e `is.na`.

In [6]:
print(colSums(is.na(df_fifa)))

           Country      Confederation          Rank_Date Points_Old_Version 
                 0               2982                  0              17415 
Points_New_Version               rank 
                 0                  0 


Considera los siguientes aspectos observados:

* Existen valores ausentes en las columnas Confederation y Points_Old_Version.
* En el caso de los valores perdidos de Points_Old_Version, destaca que los mismos parecen estar asociados a fechas en las que se utilizaban metodologías antiguas de la clasificación, lo cual hace sentido en lo referente al nombre de la columna.
* Points_New_Version presenta casos con el texto "ND", lo cual estaría ocasionando que el tipo de variable no sea el correcto. En todo caso, sucede algo similar con lo visto en el punto anterior, estos "valores ausentes" se corresponderían a fechas donde se aplicaban metodologías anteriores.

Mira más a fondo otras columnas. Por ejemplo, observa los valores posibles de Confederation usando las funciones `unique` y `count`.

In [7]:
# Observar valores de confederation con unique
unique(df_fifa["Confederation"])

Confederation
<chr>
UEFA
CONMEBOL
CONCACAF
CAF
AFC
NA


In [8]:
# Observar valores de Confederation con count
count(df_fifa, Confederation)

Confederation,n
<chr>,<int>
AFC,12481
CAF,14876
CONCACAF,9664
CONMEBOL,2860
UEFA,14934
NA,2982


Verifica aquellas selecciones nacionales que tienen valores perdidos en Confederation para comprender a qué confederación corresponderían.

In [9]:
unique(df_fifa[is.na(df_fifa["Confederation"]),]["Country"])

Country
<chr>
Australia
New Zealand
Fiji
Tahiti
Solomon Islands
Vanuatu
Papua New Guinea
Tonga
Samoa


De esta observación en particular se confirma que la confederación faltante se corresponde a la de Oceanía (OFC).

Utiliza ahora la función `summary` para generar un resumen estadístico de la columna rank. 

In [10]:
summary(df_fifa["rank"])

      rank      
 Min.   :  1.0  
 1st Qu.: 51.0  
 Median :101.0  
 Mean   :101.6  
 3rd Qu.:152.0  
 Max.   :209.0  

Notemos que en algunas mediciones se han alcanzado hasta 209 paises puesto que este es su valor máximo. De hecho, vale mencionar que la FIFA actualmente reconoce a 211 países por lo que podríamos concluir que algunos países comparten una misma posición.

Mantengamos esa curiosidad por conocer más de los datos. Cuenta con `count` cuántos paises fueron rankeados en cada una de las fechas. Ordena este conteo por fecha con la función `arrange` y muestra únicamente los primeros y últimos resultados con las funciones `slice_head` y `slice_tail` respectivamente.

In [11]:
# Mirar primeros resultados por fecha
slice_head(
    arrange(
        count(
            df_fifa, 
            Rank_Date
        ), 
        Rank_Date
    ), 
    n = 5
)

Rank_Date,n
<date>,<int>
1993-08-08,167
1993-09-23,167
1993-10-22,167
1993-11-19,168
1993-12-23,168


In [12]:
# Mirar ultimos resultados por fecha
slice_tail(
    arrange(
        count(
            df_fifa, 
            Rank_Date
        ), 
        Rank_Date
    ), 
    n = 5
)

Rank_Date,n
<date>,<int>
2018-02-15,211
2018-03-15,211
2018-04-12,211
2018-05-17,211
2018-06-07,211


Notemos que para estas consultas simples realizadas, hemos necesitado anidar funciones lo cual hace que el mismo sea más complicado de leer. **Tidyverse** permite escribir esto mediante un mécanismo más eficiente llamado **pipelines** a través del código "%>%" que genera un esquema más secuencial de instrucciones en nuestros programas. Aplícalo nuevamente para mejorar los scripts anteriores.

In [13]:
# Mirar primeros resultados por fecha
df_fifa %>%
count(Rank_Date) %>%
arrange(Rank_Date) %>%
slice_head(n = 5)

Rank_Date,n
<date>,<int>
1993-08-08,167
1993-09-23,167
1993-10-22,167
1993-11-19,168
1993-12-23,168


In [14]:
# Mirar ultimos resultados por fecha
df_fifa %>%
count(Rank_Date) %>%
arrange(Rank_Date) %>%
slice_tail(n = 5)

Rank_Date,n
<date>,<int>
2018-02-15,211
2018-03-15,211
2018-04-12,211
2018-05-17,211
2018-06-07,211


Comprueba finalmente si existen casos duplicados en el dataset. Considerando que existe una gran cantidad de registros en el dataset (aproximadamente 58 mil filas), sería muy poco práctico hacer una comprobación manual, por tanto utiliza la función `duplicated`.

In [15]:
sum(duplicated(df_fifa))

[1] 41

Mira en mayor detalle estos duplicados para saber en que consisten. Utiliza la funcion `filter` para visualizar solamente los casos.

In [16]:
df_fifa %>%
filter(duplicated(.)) %>%
count(Country)

Country,n
<chr>,<int>
Brunei Darussalam,1
Finland,1
Samoa,1
Sudan,37
United Arab Emirates,1


Gran parte de estos duplicados corresponden a calificaciones asociadas a la selección de Sudán. En general, los valores duplicados pueden considerarse errores en el registro de información y nos interesa eliminar estos casos a fin de evitar distorsiones en los futuros resultados del análisis. 

A partir de todo este entendimiento de los datos, establece un plan de acción para procesar esta información de modo que la tabla final resultante esté lista para cualquier análisis que desees hacer.

**PLAN DE ACCIÓN PARA PREPARACIÓN DE DATOS**

| # | Acción |
| ---- | ---- |
| 1 | Los nombres de las columnas no están bien presentados, en concreto rank esta en minúsculas y el resto no. Se debería estandarizar a un mismo formato que siga el esquema *snake_case* (minúsculas y con espacios representados por "_"). |
| 2 | Las columna Confederation presenta valores ausentes que pueden corregirse inputando el valor "OFC" debido a que se identificó que estos son los países a los que corresponden dichos errores en la información. |
| 3 | La columna Poins_New_Version debe cambiarse a un tipo numérico con decimales (*double*) que responde más a su naturaleza de valores numéricos. Con este propósito en primera instancia conviene dar tratamiento a los casos "ND" detectados cambiándolos por valores ausentes de tipo numérico. |
| 4 | Las columnas Points_Old_Version y Point_New_Version igualmente presentan valores ausentes, sin embargo por el contexto del caso éstos responden a los cambios metodológicos generados por la FIFA. En todo caso convendría crear una nueva columna que consolide esta información y que por tanto ya no tenga casos ausentes. |
| 5 | A partir de la columna Rank_Date, se podría contar con campos adicionales de años y meses considerando que la calificación de la FIFA mentiene generalmente esta periodicidad. |
| 6 | Existen 41 filas duplicadas que deben eliminarse considerando que son errores en el dataset y afectarían los resultados de futuros análisis. |

## Preparación de los datos

Utilizando nuestro plan de acción, procedamos a dejar listo este dataset para los análisis futuros que vamos a desarrollar. Entonces, en primer lugar ajusta los nombres de las columnas para que estén en formato *snake_case*. Recuerda que este formato exige que todos los nombres se encuentren en minúsculas y los espacios se sustituyan por "_".

In [17]:
names(df_fifa) <- names(df_fifa) %>% 
tolower() %>%
gsub(" ", "_", .)

glimpse(df_fifa)

Rows: 57,797
Columns: 6
$ country            <chr> "Germany", "Italy", "Switzerland", "Sweden", "Argen…
$ confederation      <chr> "UEFA", "UEFA", "UEFA", "UEFA", "CONMEBOL", "UEFA",…
$ rank_date          <date> 1993-08-08, 1993-08-08, 1993-08-08, 1993-08-08, 19…
$ points_old_version <dbl> 57, 57, 50, 55, 51, 54, 52, 55, 49, 51, 55, 45, 49,…
$ points_new_version <chr> "ND", "ND", "ND", "ND", "ND", "ND", "ND", "ND", "ND…
$ rank               <dbl> 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, …


Como siguiente paso, inputa el valor "OFC" a los casos ausentes de la columna confederation. Utiliza para esto las funciones `mutate` e `if_else`.

In [18]:
df_fifa <- df_fifa %>%
mutate(confederation = if_else(is.na(confederation),"OFC", confederation))

df_fifa %>% 
count(confederation)

confederation,n
<chr>,<int>
AFC,12481
CAF,14876
CONCACAF,9664
CONMEBOL,2860
OFC,2982
UEFA,14934


Ajusta ahora la columna points_new_version realizando los siguientes pasos:

1. Cambia los valores perdidos "ND" por un valor perdido adecuado usando para esto el valor `NA`.
2. Cambia el tipo de variable a *double* utilizando la función `as.double`.

In [19]:
# Cambiar valores perdidos ND por NA
df_fifa <- df_fifa %>%
mutate(points_new_version = if_else(
    points_new_version == "ND", 
    NA, 
    points_new_version
))

print(colSums(is.na(df_fifa)))

           country      confederation          rank_date points_old_version 
                 0                  0                  0              17415 
points_new_version               rank 
             40382                  0 


In [20]:
# Cambiar tipo de variable de points_new_version
df_fifa <- df_fifa %>%
mutate(points_new_version = as.double(points_new_version))

glimpse(df_fifa)

Rows: 57,797
Columns: 6
$ country            <chr> "Germany", "Italy", "Switzerland", "Sweden", "Argen…
$ confederation      <chr> "UEFA", "UEFA", "UEFA", "UEFA", "CONMEBOL", "UEFA",…
$ rank_date          <date> 1993-08-08, 1993-08-08, 1993-08-08, 1993-08-08, 19…
$ points_old_version <dbl> 57, 57, 50, 55, 51, 54, 52, 55, 49, 51, 55, 45, 49,…
$ points_new_version <dbl> NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA, NA,…
$ rank               <dbl> 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, …


Consolida las columnas points_old_version y points_new_version en una nueva columna llamada *points*, de forma que ya no existan valores ausentes por los cambios metodológicos.

In [21]:
df_fifa <- df_fifa %>%
mutate(points = if_else(
    is.na(points_old_version), 
    points_new_version, 
    points_old_version
))

print(colSums(is.na(df_fifa)))

           country      confederation          rank_date points_old_version 
                 0                  0                  0              17415 
points_new_version               rank             points 
             40382                  0                  0 


Para evitar confusiones posteriores, elimina las columnas points_old_version y points_new_version. Utiliza la función `select`.

In [22]:
df_fifa <- df_fifa %>%
select(-points_old_version, -points_new_version)

glimpse(df_fifa)

Rows: 57,797
Columns: 5
$ country       <chr> "Germany", "Italy", "Switzerland", "Sweden", "Argentina"…
$ confederation <chr> "UEFA", "UEFA", "UEFA", "UEFA", "CONMEBOL", "UEFA", "UEF…
$ rank_date     <date> 1993-08-08, 1993-08-08, 1993-08-08, 1993-08-08, 1993-08…
$ rank          <dbl> 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 1…
$ points        <dbl> 57, 57, 50, 55, 51, 54, 52, 55, 49, 51, 55, 45, 49, 42, …


Es posible extraer sus componentes fundamentales de rank_date. En este sentido, crea una columna *year* y otra *month* que contenga la información correspondiente. 

In [23]:
# Crear columna year y month
df_fifa <- df_fifa %>%
mutate(year = year(rank_date)) %>%
mutate(month = month(rank_date))

sample_n(df_fifa, size = 5)

country,confederation,rank_date,rank,points,year,month
<chr>,<chr>,<date>,<dbl>,<dbl>,<dbl>,<dbl>
Aruba,CONCACAF,2011-10-19,172,92.34,2011,10
Panama,CONCACAF,2008-09-03,96,379.00,2008,9
Tunisia,CAF,2004-11-10,36,630.00,2004,11
Syria,AFC,2007-10-24,126,256.00,2007,10
Cuba,CONCACAF,2009-10-16,76,393.00,2009,10


Adicionalmente, y si bien no está en el plan de acción, podría ser útil especificar en una columna nueva la metodología utilizada para el cálculo de puntos. Crea por tanto la columna *method* la cual cumpla con los siguientes criterios secuenciales:

* Si el año y mes de la fila es menor a Enero 1999 entonces su valor será "1993 - 1998".
* Si el año y mes de la fila es menor a Agosto 2006 entonces su valor será "1999 - 2006"
* Si el año y mes de la fila es menor a Agosto 2011 entonces su valor será "2006 - 2011"
* Todos los demás casos deberían tener el valor "2011 - 2018"

In [24]:
df_fifa <- df_fifa %>%
mutate(method = if_else(
    year * 100 + month < 199901,
    "1993-1998",
    if_else(
        year * 100 + month < 200608,
        "1999-2006",
        if_else(
            year * 100 + month < 201108,
            "2006-2011",
            "2011-2018"
        )
    )
))

sample_n(df_fifa, size = 5)

country,confederation,rank_date,rank,points,year,month,method
<chr>,<chr>,<date>,<dbl>,<dbl>,<dbl>,<dbl>,<chr>
Turks and Caicos Islands,CONCACAF,2011-05-18,193,13.00,2011,5,2006-2011
Russia,UEFA,2017-02-09,61,592.03,2017,2,2011-2018
Norway,UEFA,2001-07-18,21,660.00,2001,7,1999-2006
Vietnam,AFC,1999-05-19,110,326.00,1999,5,1999-2006
American Samoa,OFC,2016-09-15,190,63.75,2016,9,2011-2018


Finalmente, da tratamiento a los duplicados eliminándolos con el método `drop_duplicates`.

In [25]:
df_fifa <- df_fifa %>%
filter(!duplicated(.))

sum(duplicated(df_fifa))

[1] 0

Listo, hemos concluido con nuestro plan de acción. Dale una mirada nuevamente al dataframe limpio extrayendo una muestra de 15 elementos.

In [26]:
sample_n(df_fifa, size = 15)

country,confederation,rank_date,rank,points,year,month,method
<chr>,<chr>,<date>,<dbl>,<dbl>,<dbl>,<dbl>,<chr>
St Kitts and Nevis,CONCACAF,1999-01-27,137,17.00,1999,1,1999-2006
Tunisia,CAF,1998-08-19,26,54.00,1998,8,1993-1998
New Zealand,OFC,2008-03-12,91,349.00,2008,3,2006-2011
Congo,CAF,2008-05-07,93,349.00,2008,5,2006-2011
Grenada,CONCACAF,2014-07-17,142,181.53,2014,7,2011-2018
Antigua and Barbuda,CONCACAF,2010-04-28,119,217.00,2010,4,2006-2011
Bahrain,AFC,2014-11-27,125,243.34,2014,11,2011-2018
Denmark,UEFA,2011-03-09,28,802.00,2011,3,2006-2011
St Vincent and the Grenadines,CONCACAF,2011-06-29,154,116.00,2011,6,2006-2011


## Análisis de datos

Es hora que realices tu primer análisis de datos ahora que tienes una tabla con información limpia. Para un correcto análisis vale que tengas siempre en cuenta lo siguiente:

* Su propósito fundamental es **responder preguntas de negocio**, por lo que siempre debes buscar dar soluciones concretas en base a tus resultados.
* No se limita a realizar gráficos o tablas resumen, sino a generar **conclusiones** que aporten valor a partir de estas visualizaciones.
* Debe mantener un **hilo de ideas** conductor lo cual implica orden y consistencia entre los resultados.

Visto esto, a continuación verás algunas preguntas de negocio a responder: 

### ¿Cómo ha sido el comportamiento anual del puntaje de la CONMEBOL a partir de la última metodología?

En primer lugar identifica entre las columnas de la tabla cuál es la métrica relevante que debes obtener:

`A partir de la columna points, la métrica sería el puntaje promedio por año.`

Ahora, especifica si debes realizar filtros en la tabla a partir de algunas condiciones:

`Se debe filtrar la columna confederation manteniendo solamente las filas de "CONMEBOL". Además se debe filtrar la culumna method manteniendo solamente las filas de "2011-2018".`

Para terminar, indica si debes agregar o resumir los datos por alguna de las columnas: 

`Los datos deben agregarse por la columna year.`

Construye el código necesario para responder la pregunta utilizando como criterios los aspectos contestados antes.

In [27]:
puntos_conmebol <- df_fifa %>%
# Filtrar datos
filter(confederation == "CONMEBOL") %>%
filter(method == "2011-2018") %>%
# Agrupar datos
group_by(year) %>%
# Metrica a utilizar
summarise(points_avg = mean(points)) %>%
ungroup()

puntos_conmebol

year,points_avg
<dbl>,<dbl>
2011,818.3670
2012,848.1707
2013,891.1479
2014,920.0970
2015,912.7798
2016,996.9699
2017,1035.5418
2018,959.8545


Genera dos conclusiones relevantes. Recuerda que una conclusión no se limita a una descripción de lo que obtuviste, sino a implicancias, hipótesis o recomendaciones a partir de estos resultados que aporten a la toma de decisiones.

**CONCLUSIONES**

| # | Conclusión |
| ---- | ---- |
| 1 | Se observa un crecimiento sostenido en el puntaje de las selecciones de CONMEBOL en este periodo lo cual está asociado al mejoramiento de selecciones no tan protagonistas como son Chile y Ecuador. En concreto, esto se confirma con las Copas América ganadas por Chile en los años 2015 y 2016, así como con los éxitos de clubes ecuatorianos a nivel internacional como IDV y LDU, los cuales son la base actual del seleccionado nacional. |
| 2 | A pesar de esto, en el último año se evidencia una reducción significativa en el puntaje medio de CONMEBOL, explicado seguramente por los malos resultados de varias selecciones de esta confederación previo al mundial de Rusia de 2018: Argentina perdió dos partidos importantes en su etapa preparatoria (2 - 4 versus Nigeria y 1 - 6 versus España); Colombia perdió un partido preparatorio con Corea del Sur por (1 - 2) y empató otro contra Australia (0 - 0); Chile y Ecuador quedaron eliminadas en la fase clasificatoria a pesar de estar bien posicionadas y en constante crecimiento los últimos años. Bolivia recibió sanciones por parte de la FIFA producto de asuntos administrativos en su ente federativo nacional. |

### ¿Cual ha sido la confederación peor evaluada en los últimos 3 años registrados?

Establece cuál es la métrica de análisis, los filtros y agrupamientos que necesitas y da una respuesta a la pregunta con tu código.

In [28]:
puntos_x_conf <- df_fifa %>%
filter(year >= 2016) %>%
group_by(confederation, year) %>%
summarise(points = mean(points)) %>%
ungroup() %>%
pivot_wider(
    names_from = year,
    values_from = points
)

puntos_x_conf

`summarise()` has grouped output by 'confederation'. You can override using the
`.groups` argument.


confederation,2016,2017,2018
<chr>,<dbl>,<dbl>,<dbl>
AFC,252.82580,259.3869,258.7086
CAF,374.93177,367.8040,341.1719
CONCACAF,290.68317,278.1365,253.5680
CONMEBOL,996.96992,1035.5418,959.8545
OFC,98.86447,110.2392,105.0944
UEFA,644.86955,642.8007,647.5584


Genera una conclusión relevante.

**CONCLUSIONES**

| # | Conclusión |
| ---- | ---- |
| 1 | OFC tiene el peor posicionamiento entre las confederaciones lo cual se explicaría por el bajo nivel de fútbol competitivo y profesional que existe en los paises de Oceanía. Esto se acrecentó incluso más el año 2006 en el que Australia se afilió a la AFC para mantener un mayor nivel en sus enfrentamientos deportivos. |